### PR TITLE
chore(litmus-portal): Add unit testing setup using `@testing-library/react`

### PR DESCRIPTION
**Pull request by saswatamcode**
## Proposed changes

This PR adds in the unit testing setup using `@testing-library/react` with `jest` as the test runner. Also, adds in two sample tests for `ReduxRoot` and `ChooseWorkflow` components. 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Special notes for your reviewer:
Currently, `i18n` does not work with tests due to issues. Will be added in later.

### CODE REVIEW

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  preset: 'ts-jest/presets/js-with-ts',
+  testMatch: [
+    '<rootDir>/src/**/__tests__/**/*.{js,jsx,ts,tsx}',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/jest.config.js:4]: we are using create react app...which already have jest config I think....is this config required then?
Try to remove this and check if everything works the same or not.

File: litmus-portal/frontend/src/ReduxRoot.test.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import ReduxRoot from './ReduxRoot';
+import '@testing-library/jest-dom';
+
+describe('<ReduxRoot />', () => {
+  it('renders', async () => {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/ReduxRoot.test.tsx:7]: This is not a useful test...for trying out it's fine but should be removed before merge...
Try running test coverage and look for untested logic...if you want a real testcase

File: litmus-portal/frontend/src/ReduxRoot.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ const ReduxRoot = () => {
     <ApolloProvider client={client}>
       <Provider store={store}>
         <PersistGate persistor={persistor}>
-          <App />
+          <App data-testid="app" />
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/ReduxRoot.tsx:22]: this also won't be required as the test case for App didn't make sense

File: litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/__tests__/ChooseWorkflow.test.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { render } from '../../../../testHelpers/test-util';
+import ChooseWorkflow from '../index';
+import '@testing-library/jest-dom';
+
+describe('ChooseWorkflow', () => {
+  it('Renders header, <RadioGroup>, <Accordion', () => {
+    const { getByTestId } = render(<ChooseWorkflow />);
+    expect(getByTestId('chooseworkflow-header')).toBeTruthy();
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/__tests__/ChooseWorkflow.test.tsx:9]: try to write the test case with actual states to check the logic

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/ReduxRoot.test.tsx (hunk 2 above)
Comment (saswatamcode) [reply litmus-portal/frontend/src/ReduxRoot.test.tsx:7]: Yes, this was for trying out!

File: litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/__tests__/ChooseWorkflow.test.tsx (hunk 4 above)
Comment (saswatamcode) [reply litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/__tests__/ChooseWorkflow.test.tsx:9]: Sure, will try to incorporate.

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by S-ayanide — APPROVED** [review]

**Pull request merged**